### PR TITLE
feat: Add Home deep link intent

### PR DESCRIPTION
**Pull request by dcalhoun**
## Description

Support deep linking to the default tab of the application, which is useful for generic promotions found on web pages. This was added in support of https://github.com/Automattic/wp-calypso/pull/68062. Specifically, this intent is [needed for the new banner](https://github.com/Automattic/wp-calypso/pull/68062/files/6e160a2042cf4fbf6d30c68b66aff2958d685657#r976689110) displayed.

## Testing

With this branch checked out and the WordPress/Jetpack app running in an emulator, run the following command in Android Studio's terminal to confirm the custom scheme works to open the correct section: 

- **Jetpack:** `adb shell am start -a android.intent.action.VIEW -d "jetpack://home"`
- **WordPress:** `adb shell am start -a android.intent.action.VIEW -d "wordpress://home"`

For logged in users, the _My Site_ tab should display. For logged out users, the _Login Prologue_ should display. 

## Regression Notes
1. Potential unintended areas of impact
    Other deep links could stop working.
2. What I did to test those areas of impact (or what existing automated tests I relied on)
    Verified automated tests pass. Manually tested existing deep links via `adb` commands similar to the above.
3. What automated tests I added (or what prevented me from doing so)
    Tests mirroring existing handler tests.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by dcalhoun** [comment]
Hey @zwarm. 👋🏻 I noticed you have reviewed similar code changes recently. Would you be willing to review this PR please? 🙇🏻

**Review by zwarm — APPROVED** [review]
@dcalhoun - The home deep link launches the app as expected. 👍 

❓ What is the expected behavior for the not-yet logged in user? 
My experience was as follows:
- Tapped on deep link
- The app is launched
- Prompted to log in
- Upon login I am shown the "old" site selection view - which doesn't let you select a site (see attached screen), but rather you have to tap done
- Taken to my `initial screen` (which happens to be the Menu tab) for the first site on my sites list

I am going to approve the PR because it works as described; however I will leave it up to you for merging if the existing "logged out" behavior is what is expected. Thanks 🎉

**Comment by dcalhoun** [comment]
> What is the expected behavior for the not-yet logged in user? My experience was as follows:

@zwarm great question. I think it makes most sense to have the app launch displaying the _Login Prologue_. I will explore how we might accomplish that, but please feel free to share any code reference you think may be relevant. It could be leveraging `ActivityLauncher.viewMySiteInNewStack` is not the most appropriate approach. 

<details><summary>Screenshot of Login Prologue</summary>

Screen Shot  of Login Prologue

</details>

File: WordPress/src/main/java/org/wordpress/android/ui/deeplinks/DeepLinkingIntentReceiverViewModel.kt
Comment (dcalhoun) [inline WordPress/src/main/java/org/wordpress/android/ui/deeplinks/DeepLinkingIntentReceiverViewModel.kt:73]: > This condition is too complex (4). Defined complexity threshold for conditions is set to '4' [ComplexCondition]

@zwarm this was refactored to avoid the above lint warning. As a(n) Kotlin/Android novice, I welcome feedback on the refactor, if you have any.

**Comment by dcalhoun** [comment]
@zwarm I updated the implementation and testing instructions based upon your question regarding the logged out experience. I also refactored code to more closely align with existing implementations. Please let me you know what you think. 🙇🏻

**Comment by zwarm** [comment]
@dcalhoun - The refactoring changes look great. Thanks for making them ❤️ 

I did run into an unexpected situation. 

If I only have the Jetpack app installed on the device, running `adb shell am start -a android.intent.action.VIEW -d "wordpress://home"` launches the Jetpack app. If I have both apps installed, I get a "chooser" bottom sheet asking me to chose between Jetpack or WordPress.

This begs an interesting question: 
Should we be "excluding" the "wordpress" URI deeplink schemes when building Jetpack? The WordPress `AndroidManifest.xml` is the base xml and we layer Jetpack over it. I wonder if we should completely override the entire `DeepLinkingIntentReceiverActivity` section to avoid these types of issues? 

Or does this even matter? This would be the case will all schemes for WordPress, not just the new one you added. 

wdyt?

**Comment by dcalhoun** [comment]
> If I only have the Jetpack app installed on the device, running `adb shell am start -a android.intent.action.VIEW -d "wordpress://home"` launches the Jetpack app. If I have both apps installed, I get a "chooser" bottom sheet asking me to chose between Jetpack or WordPress.
>
> This begs an interesting question: Should we be "excluding" the "wordpress" URI deeplink schemes when building Jetpack? The WordPress `AndroidManifest.xml` is the base xml and we layer Jetpack over it. I wonder if we should completely override the entire `DeepLinkingIntentReceiverActivity` section to avoid these types of issues?
> 
> Or does this even matter? This would be the case will all schemes for WordPress, not just the new one you added.

@zwarm great find and question. My initial response is that this behavior is expected given (1) the shared manifests you referenced and (2) the intended positioning of Jetpack as an "upgraded WordPress experience." I would expect both the Jetpack and WordPress apps to offer handling `wordpress://` schemes. 

However, I could understand a counter argument that long term these app-specific URI schemes should be bifurcated, and that the only WordPress-centric deep links the Jetpack app should handle are ones based upon web domains, e.g. `wordpress.com`. 

This is likely a subject worth discussing with the wider team at a higher level. 

That said, I do not believe this issue is specific to this new deep link intent and should not block this PR. WDYT?

**Comment by zwarm** [comment]
@dcalhoun 
> This is likely a subject worth discussing with the wider team at a higher level.

100% agree. Go ahead and bring this up in slack or a P2

> That said, I do not believe this issue is specific to this new deep link intent and should not block this PR. WDYT?

Also agree with this. I will merge!!
🥳

**Comment by dcalhoun** [comment]
> If I only have the Jetpack app installed on the device, running `adb shell am start -a android.intent.action.VIEW -d "wordpress://home"` launches the Jetpack app. If I have both apps installed, I get a "chooser" bottom sheet asking me to chose between Jetpack or WordPress.

@zwarm after researching Intent URI schemes a bit further, I believe I was mistaken with my original testing instructions. 

**Provided:** 
* `adb shell am start -a android.intent.action.VIEW -d "jetpack://home"`
* `adb shell am start -a android.intent.action.VIEW -d "wordpress://home"`

**Intended:** 
* `adb shell 'am start "intent://home/#Intent;scheme=jetpack;package=com.jetpack.android.beta;end"'`
* `adb shell 'am start "intent://home/#Intent;scheme=wordpress;package=org.wordpress.android.beta;end"'`

The _Intended_ URI scheme is the [one found within Calypso banners](https://github.com/Automattic/wp-calypso/blob/5c0a7d68ae0f31704c87f80c8f3065068f8c837e/client/blocks/app-banner/index.jsx#L139). From my testing now, using the _Intended_ URI scheme does not result in Android presenting a "chooser" when both the WordPress and Jetpack apps are installed. 

It may still be worth discussing whether the Jetpack app should handle `wordpress://` URI schemes — which I will publish — but it seems it may be less impactful for the current web banners.

**Comment by zwarm** [comment]
> Intended URI scheme does not result in Android presenting a "chooser" when both the WordPress and Jetpack apps are installed.

Yep, sounds like the correct behavior when the package name is set explicitly. 

🤔 Makes wonder if we ever use the "jetpack://" or "wordpress://" intents without the package names and does it really matter if JP can open the WP links? I guess it all depends on the experience we want our users to encounter. 

Thanks for the added context.

**Pull request merged**
